src/Microsoft.Identity.Web.TokenAcquisition/MsAuth10AtPop.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Identity.Web
+{
+    internal static class MsAuth10AtPop
+    {
+        public static AcquireTokenForClientParameterBuilder WithAtPop(
+            this AcquireTokenForClientParameterBuilder builder,
+            X509Certificate2 clientCertificate,
+            string popPublicKey,
+            string jwkClaim,
+            string clientId)
+        {
+            builder.WithProofOfPosessionKeyId(popPublicKey);
+            builder.OnBeforeTokenRequest((data) =>
+             {
+                 string? signedAssertion = GetSignedClientAssertion(
+                     clientCertificate,
+                     data.RequestUri.AbsoluteUri,
+                     jwkClaim,
+                     clientId);
+
+                 data.BodyParameters.Remove("client_assertion");
```

Comment:
@bgavrilMS 
Would it be possible, when we use builder.WithProofOfPosessionKeyId(popPublicKey) that MSAL does not compute the signed assertion? (as it's not necessary)
I can create a GitHub issue in the MSAL.NET repo if you agree.